File: src/Shared/Polyrific.Catapult.Shared.ApiClient/ExternalServiceTypeService.cs
// Copyright (c) Polyrific, Inc 2018. All rights reserved.

using System.Collections.Generic;
using System.Threading.Tasks;
using Polyrific.Catapult.Shared.Dto.ExternalServiceType;
using Polyrific.Catapult.Shared.Service;

namespace Polyrific.Catapult.Shared.ApiClient
{
    public class ExternalServiceTypeService : BaseService, IExternalServiceTypeService
    {
        public ExternalServiceTypeService(IApiClient api) : base(api)
        {
        }

        public async Task<ExternalServiceTypeDto> GetExternalServiceType(int id)
        {
            var path = $"serviceType/{id}";

            return await Api.Get<ExternalServiceTypeDto>(path);
        }

        public async Task<ExternalServiceTypeDto> GetExternalServiceTypeByName(string name)
        {
            var path = $"serviceType/name/{name}";

            return await Api.Get<ExternalServiceTypeDto>(path);
        }

        public async Task<List<ExternalServiceTypeDto>> GetExternalServiceTypes()
        {
            var path = "serviceType";

            return await Api.Get<List<ExternalServiceTypeDto>>(path);
        }
    }
}

File: src/Web/Polyrific.Catapult.Web/ClientApp/src/app/core/services/external-service-type.service.ts
import { Injectable } from '@angular/core';
import { ApiService } from './api.service';
import { ExternalServiceTypeDto } from '../models/external-service/external-service-type-dto';

@Injectable()
export class ExternalServiceTypeService {

  constructor(private api: ApiService) { }

  getExternalServiceType(id: number) {
    return this.api.get<ExternalServiceTypeDto>(`serviceType/${id}`);
  }

  getExternalServiceTypeByName(name: string) {
    return this.api.get<ExternalServiceTypeDto>(`serviceType/name/${name}`);
  }

  getExternalServiceTypes(includeProperties: boolean) {
    let path = 'serviceType';

    if (includeProperties) {
      path += '?includeProperties=true';
    }

    return this.api.get<ExternalServiceTypeDto[]>(path);
  }
}
